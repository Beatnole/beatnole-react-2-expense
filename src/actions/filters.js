export const setTextFilter = (text = '') => ({
    type: 'SET_TEXT_FILTER',
    text
})

export const setSortByAmount = () => ({
    type: 'SET_SORT_BY_AMOUNT'
})

export const setSortByDate = () => ({
    type: 'SET_SORT_BY_DATE'
})

export const setStartDate = (startDate) => ({
    type: 'SET_START_DATE',
    startDate
})

export const setEndDate = (endDate) => ({
    type: 'SET_END_DATE',
    endDate
})