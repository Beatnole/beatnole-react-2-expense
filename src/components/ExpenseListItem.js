import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom';
import moment from 'moment'


 export const ExpenseListItem = ({ description, amount, createdAt, id }) => (
    <div>
        <Link to={`/edit/${id}`}>    
            <h3>{description}</h3>
        </Link>
            <p>{`$${(amount) / 100}`} - {createdAt}</p>
    </div>
)

export default connect()(ExpenseListItem)

