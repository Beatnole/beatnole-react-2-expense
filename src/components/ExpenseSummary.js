import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import { getExpensesTotal } from '../selectors/expense-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal, allExpenseCount }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses' ;
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  
  return (
    <div>
      <h1>Viewing {expenseCount} {expenseWord} of {allExpenseCount} totalling {formattedExpensesTotal}</h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  const allExpenses = selectExpenses(state.expenses, { text: '', sortBy: 'date' , startDate: 0, })

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: getExpensesTotal(visibleExpenses),
    allExpenseCount: allExpenses.length
  };
};

export default connect(mapStateToProps)(ExpensesSummary);