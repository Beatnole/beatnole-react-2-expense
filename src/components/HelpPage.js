import React from 'react';


const HelpPage = () => (
    <div>
        This is from the helpPage
    </div>
)

export default HelpPage