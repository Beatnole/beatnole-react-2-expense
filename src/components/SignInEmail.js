import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { firebase } from '../firebase/firebase'
import { Link } from 'react-router-dom'

const SignInPage = ({ history }) =>
  <div className="box-layout">
    <div className="box-layout__box">
         <div className="form">
            <h1>Expensify Sign In</h1>
            <SignInForm history={history} />
            <Link className="button button-password-reset" to="/forgotEmail">Forgot Password</Link>
        </div>
    </div>
  </div>

  const byPropKey = (propertyName, value) => () => ({
  [propertyName]: value,
});

const INITIAL_STATE = {
  email: '',
  password: '',
  error: null,
};

class SignInForm extends Component {
  constructor(props) {
    super(props);

    this.state = { ...INITIAL_STATE };
  }

  onSubmit = (event) => {
    const {
      email,
      password,
    } = this.state;

    const {
      history,
    } = this.props;

    firebase.auth().signInWithEmailAndPassword(email, password)
      .then(() => {
        this.setState({ ...INITIAL_STATE });
        history.push('/dashboard')
      })
      .catch(error => {
        this.setState(byPropKey('error', error));
      });

    event.preventDefault();
  }

  render() {
    const {
      email,
      password,
      error,
    } = this.state;

    const isInvalid =
      password === '' ||
      email === '';

    return (
      <form onSubmit={this.onSubmit} className="form">
        <input
          autoFocus
          className="text-input"
          value={email}
          onChange={event => this.setState(byPropKey('email', event.target.value))}
          type="text"
          placeholder="Email Address"
        />
        <input
          className="text-input"
          value={password}
          onChange={event => this.setState(byPropKey('password', event.target.value))}
          type="password"
          placeholder="Password"
        />
        <button disabled={isInvalid}
         type="submit" 
         className={isInvalid ? "button button--secondary" : "button"} >
          Sign In
        </button>

        { error && <p>{error.message}</p> }
      </form>
    );
  }
}

export default withRouter(SignInPage);

export {
  SignInForm,
};