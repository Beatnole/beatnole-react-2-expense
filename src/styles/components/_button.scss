.button {
    background: $dark-blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    margin-top: $m-size;
    padding: $s-size;
    text-decoration: none;
    text-align: center;
}
.button--link {
    background: none;
}

.button--secondary {
    background: #888888;
}

.button-password-reset {
    background: $dark-red;
    
    margin-top: $s-size;
}