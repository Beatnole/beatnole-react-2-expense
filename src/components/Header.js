import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux'
import { startSignOut } from '../actions/auth'

export const Header = ({ startSignOut }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
        <button onClick={startSignOut}>Logout</button>
        </header>
)

const mapDispatchToProps = (dispatch) => ({
    startSignOut: () => dispatch(startSignOut())
})

export default connect(undefined, mapDispatchToProps)(Header)