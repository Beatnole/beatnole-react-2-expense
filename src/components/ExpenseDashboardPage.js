import React from 'react';
import Header from './Header'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummary from './ExpenseSummary'

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export default ExpenseDashboardPage