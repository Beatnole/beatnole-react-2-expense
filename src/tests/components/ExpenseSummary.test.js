import React from 'react';
import { ExpenseSummary } from '../../components/ExpenseSummary'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'
import { filters, altFilters } from '../fixtures/filters';

let setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate, wrapper;

beforeEach(() => {
  setTextFilter = jest.fn();
  sortByDate = jest.fn();
  sortByAmount = jest.fn();
  setStartDate = jest.fn();
  setEndDate = jest.fn();
  wrapper = shallow(
    <ExpenseSummary
      filters={filters}
      setTextFilter={setTextFilter}
      sortByDate={sortByDate}
      sortByAmount={sortByAmount}
      setStartDate={setStartDate}
      setEndDate={setEndDate}
    />
  );
});

test('ExpenseSummary Display with 3 expenses', () => {
    expect(wrapper).toMatchSnapshot();
  });

