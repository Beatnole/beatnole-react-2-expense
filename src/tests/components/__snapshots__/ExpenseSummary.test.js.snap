// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseSummary Display with 3 expenses 1`] = `undefined`;
